src/transaction.rs

```diff
@@ -147,36 +188,26 @@ pub fn handle_blocks(
     let ser_comp_note = serialize_comp_note(comp_note).map_err(|_| "couldn't decrypt notes")?;
     let ser_new_comp_note = serialize_comp_note(new_notes).map_err(|_| "couldn't decrypt notes")?;
 
-    let mut ser_nullifiers: Vec<String> = Vec::with_capacity(nullifiers.len());
-    for nullif in nullifiers.iter() {
-        ser_nullifiers.push(hex::encode(nullif.0));
-    }
-
     let mut buff = Vec::new();
     tree.write(&mut buff)
         .map_err(|_| "Cannot write tree to buffer")?;
 
     Ok(serde_wasm_bindgen::to_value(&JSTxSaplingData {
         decrypted_notes: ser_comp_note,
-        nullifiers: ser_nullifiers,
+        nullifiers,
```

Comment:
for  a future PR we can avoid returning nullifiers and remove the spent notes directly in the handle_blocks